lib/streams.js

```diff
@@ -27,6 +28,16 @@ function createCoreStream (db, view) {
   return ite
 }
 
+function createDiscoveryKeyStream (db, view, namespace) {
+  const start = namespace ? store.coreByAliasStart(namespace) : store.coreStart()
+  const end = namespace ? store.coreByAliasEnd(namespace) : store.coreEnd()
+
+  const ite = view.iterator(db, start, end, false)
+
+  ite._readableState.map = (data) => mapAllDiscoveryKeys(data, namespace)
```

Comment:
use two static functions instead and switch here on namespace for which one to set